src/d3a/models/strategy/storage.py

```diff
@@ -388,6 +388,8 @@ def event_balancing_market_cycle(self):
                                                                               self.owner.name)
 
     def buy_energy(self, market):
+        if self.state.has_battery_reached_max_power(-0.0001, market.time_slot):
```

Comment:
Wouldn'T it be better to use a FLOATING_POINT_TOLERANCE then a constant value here ?

Reply:
Indeed, fixing this